### ISSUE TITLE
Torchio Transformation Does not Working properly

### ISSUE THREAD

**Issue by shathaa1983** [fepegar/torchio#1016]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Problem summary

I tried to apply Torchio preprocessing to my dataset (Prostate MRI) images. I checked the volume for both (image, mask) and I found that each image's volume match the mask's volume. However, after I applied tio.compose I got an error :

![Screenshot 2023-01-19 at 16-24-02 ProstateX-JMRI-Torchio_3DUNET_(128,128,32)_Resample(0 5,0 5,3)_4bloks - Jupyter Notebook](https://user-images.githubusercontent.com/69587655/213565481-3cbb8e20-ebbb-47ac-ad65-48f5423cf200.png)

### Code for reproduction

```python
training_transform = tio.Compose([
    tio.ToCanonical(),
    tio.Resample((0.5, 0.5, 3)),
    tio.CopyAffine('prostate'),
    tio.CropOrPad((128, 128, 32)),
    tio.RandomMotion(p=0.2),
#     tio.HistogramStandardization({'prostate': landmarks}),
    tio.RandomBiasField(p=0.3),
    tio.ZNormalization(masking_method=tio.ZNormalization.mean),
    tio.RandomNoise(p=0.5),
    tio.RandomFlip(),
    tio.OneOf({
        tio.RandomAffine(): 0.8,
        tio.RandomElasticDeformation(): 0.2,
    }),
    tio.OneHot(),
])

testing_transform = tio.Compose([
    tio.ToCanonical(),
    tio.CopyAffine('prostate'),
    tio.Resample((0.5, 0.5, 3)),
    tio.CropOrPad((128, 128, 32)),
#     tio.HistogramStandardization({'prostate': landmarks}),
    tio.ZNormalization(masking_method=tio.ZNormalization.mean),
    tio.OneHot(),
])

train_dataset = tio.SubjectsDataset(train_data, transform=training_transform)
val_dataset = tio.SubjectsDataset(valid_data, transform=testing_transform)
test_dataset = tio.SubjectsDataset(test_data, transform=testing_transform)
print ('train_dataset =',len(train_dataset))
print ('val_dataset =',len(val_dataset))
print ('test_dataset =',len(test_dataset))
```

### Actual outcome

all data must have the same image and mask size, orientation, and spacing

### Expected outcome

all data must have the same image and mask size, orientation, and spacing

**Comment by hsyang1222** [comment]
you can find the related issue : https://github.com/fepegar/torchio/discussions/489

in your case, you can deal with it by

```python
training_transform = tio.Compose([
    tio.ToCanonical(),
    tio.Resample('prostate'),                                # this two line is key
    tio.Resample(target='prostate', exclude=['prostate']),   # this two line is key
    tio.Resample((0.5, 0.5, 3)),
```

**Comment by shathaa1983** [comment]
Thank you for your quick response. The issue is solved.
